main/streams/streams.c

```diff
@@ -1354,8 +1354,13 @@ PHPAPI int _php_stream_seek(php_stream *stream, zend_off_t offset, int whence)
 
 		switch(whence) {
 			case SEEK_CUR:
-				offset = stream->position + offset;
-				whence = SEEK_SET;
+				ZEND_ASSERT(stream->position >= 0);
+				if (UNEXPECTED(offset > ZEND_LONG_MAX - stream->position)) {
+					offset = ZEND_LONG_MAX;
+				} else {
```

Comment:
Long term we probably should warn about those large offsets

Reply:
Yeah, although it's somewhat to determine what is too large. Something like `ZEND_LONG_MAX` - chunk size may work. (I'm mostly thinking about 32bit systems.)